tests/st/harness/core/test_runner.py

```
@@ -406,6 +797,104 @@ def _schedule_exec_after_golden(
+    # batch must never mix platforms (a mixed --platform=a2a3,a5 run would
+    # otherwise execute a5 artifacts under an a2a3 device context). Bucket pending
+    # artifacts per resolved platform and chunk within each bucket.
+    pending: dict[str, list[tuple[str, Path, str]]] = {}  # platform → (cache_key, work_dir, platform)
+
+    # Consume in completion order: fill a batch as cases finish compiling and
+    # submit it immediately, so execution and compilation overlap.
+    for cfut in as_completed(key_by_fut):
+        try:
+            artifact = cfut.result()
+        except Exception:  # noqa: BLE001 — run() re-raises the compile crash with context
+            continue
```

Ruff (0.15.20)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)